### PR TITLE
plplot 5.11.0

### PR DESCRIPTION
**Pull request by tschoonj**
version bump
formula cleanup

### CODE REVIEW

File: Library/Formula/plplot.rb [hunk 1]
```diff
@@ -13,201 +11,62 @@ class Plplot < Formula
...
-  depends_on :x11 => :optional
+  option "with-java"
 
-  option 'with-java'
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "pango"
+  depends_on "libtool" => :run
+  depends_on "freetype"
+  depends_on :x11 => :recommended
```
Comment (DomT4) [inline Library/Formula/plplot.rb]: @mikemcquaid may object to making this `:recommended`. I suspect he made this `:optional`.

**Comment by tschoonj** [comment]
the funny thing is that the `:x11` option itself is actually irrelevant here: the cmake script finds the X11 headers without using `pkg-config`. So even though it was optional before, the X11 device called `xwin` was always built.

The only way to disable the X11 device is through a cmake option I now included properly here. The `xcairo` device no longer works due to the cairo to quartz transition.

I would advise to make `x11` recommended, as this way it keeps closest to the previously provided devices.
Another reason is the package `gnudatalanguage` in homebrew-science, which relies on the `xwin` device. 

By the way, as soon as this update is merged in I will update `gnudatalanguage` too as it needs additional patching to restore compatibility with this version of `plplot`.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by tschoonj** [comment]
@DomT4 Could you give this another go? The testbot had some internal error...

Thanks!

**Comment by DomT4** [comment]
It's green now. Would like @mikemcquaid's yay/nay on the X11 change before merge.

**Comment by tschoonj** [comment]
Hey @mikemcquaid could you have a look at this?

Thanks!

File: Library/Formula/plplot.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/plplot.rb]: Yes, please make this `:optional`

**Comment by tschoonj** [comment]
@mikemcquaid done!

File: Library/Formula/plplot.rb [hunk 2]
```diff
@@ -13,201 +11,62 @@ class Plplot < Formula
     sha256 "39b18e81232ae3987018662035c49a441ddd0f0fcbe9d7534620d512b7f2910c" => :mountain_lion
   end
 
-  depends_on 'cmake' => :build
-  depends_on 'pkg-config' => :build
-  depends_on 'pango'
-  depends_on :x11 => :optional
+  option "with-java"
```
Comment (mikemcquaid) [inline Library/Formula/plplot.rb]: For a future PR: might be nice to `depends_on :java => :optional` instead

**Comment by mikemcquaid** [comment]
Thanks @tschoonj!

**Pull request closed without merging**